subworkflows/local/fastq_input_check.nf

```diff
@@ -84,12 +84,17 @@ def create_fastq_channels(LinkedHashMap col) {
             error "ERROR: Please check input samplesheet -> Index read FastQ file does not exist!\n${col.filename_I1}"
         }
         array = [ meta, [ file(col.filename_R1), file(col.filename_R2), file(col.filename_I1) ] ]
+    }
+    if (params.library_generation_method == "trust4") {
+        meta.barcode_read = params.barcode_read
+        if (params.umi_read == "") {
+            meta.umi_read = null
+        } else {
+            meta.umi_read = params.umi_read
+        }
+        array = [ meta, [ file(col.filename_R1), file(col.filename_R2) ] ]
     } else {
-
         array = [ meta, [ file(col.filename_R1), file(col.filename_R2) ] ]
-        if (params.index_file) {
```

Comment:
@mapo9 this part was the one causing the other processes to fail, I fixed it now